core/domain/services/registration/form/v1/RegFormHandler.php

```diff
@@ -0,0 +1,327 @@
+        if ($registration instanceof EE_Registration) {
+            return true;
+        }
+        EE_Error::add_error(
+            esc_html__(
+                'An invalid Registration object was discovered when attempting to process your registration information.',
+                'event_espresso'
+            ),
+            __FILE__,
+            __FUNCTION__,
+            __LINE__
+        );
+        return false;
+    }
+
+
+    /**
+     * @param EE_Registration $registration
+     * @param string          $reg_url_link
+     * @param array[][]       $reg_form_data
+     * @return bool
+     * @throws EE_Error
+     * @throws ReflectionException
+     */
+    private function processRegistration(
+        EE_Registration $registration,
+        string $reg_url_link,
+        array $reg_form_data
+    ): bool {
+        $this->registrant_data->initializeRegistrantData($registration);
+        if (! $this->processRegFormData($registration, $reg_url_link, $reg_form_data)) {
+            return false;
+        }
+        // RegFormAttendeeFactory
+        if (! $this->attendee_factory->create($registration, $reg_url_link)) {
+            return false;
+        }
+        // at this point, we should have enough details about the registrant to consider the registration
+        // NOT incomplete
+        $this->registration_processor->toggle_incomplete_registration_status_to_default(
+            $registration,
+            false,
+            new Context(
+                'spco_reg_step_attendee_information_process_registrations',
+                esc_html__(
+                    'Finished populating registration with details from the registration form after submitting the Attendee Information Reg Step.',
+                    'event_espresso'
+                )
+            )
+        );
+        // we can also consider the TXN to not have been failed, so temporarily upgrade it's status to
+        // abandoned
+        $this->checkout->transaction->toggle_failed_transaction_status();
+        // if we've gotten this far, then let's save what we have
+        $registration->save();
+        // add relation between TXN and registration
+        $this->associateRegistrationWithTransaction($registration);
+        return true;
+    }
+
+
+    /**
+     * @param EE_Registration $registration
+     * @param string          $reg_url_link
+     * @param array           $reg_form_data
+     * @return bool
+     * @throws EE_Error
+     * @throws ReflectionException
+     */
+    private function processRegFormData(EE_Registration $registration, string $reg_url_link, array $reg_form_data): bool
+    {
+        if (isset($reg_form_data[ $reg_url_link ])) {
+            // do we need to copy basic info from primary attendee ?
+            $copy_primary = isset($reg_form_data[ $reg_url_link ]['additional_attendee_reg_info'])
+                                  && absint($reg_form_data[ $reg_url_link ]['additional_attendee_reg_info']) === 0;
```

Comment:
Why not simply this?
```suggestion
            $copy_primary = empty($reg_form_data[ $reg_url_link ]['additional_attendee_reg_info']);
```

Reply:
i'm guessing (since that code was probably originally written close to a decade ago) that `additional_attendee_reg_info` might hold value with different data types... like `bool`, `int`, `string` so running `absint()` on a string value (and then comparing against `0`) is very different than just using `empty()`.

what you've suggested might very well work fine, but i'm really hesitant to modify the logic in older code if I don't have to because there may be some weird edge case that some seemingly unnecessary line of code is accounting for. I've tried really hard to keep things operating exactly the same throughout this refactor because I don't want the existing code to get broken :\